src/schema/schema_drop.c

```diff
@@ -234,8 +236,10 @@ __drop_tiered(WT_SESSION_IMPL *session, const char *uri, bool force, const char
         WT_ERR(__wt_tiered_name(session, &tiered->iface, i, WT_TIERED_NAME_OBJECT, &name));
         __wt_verbose(session, WT_VERB_TIERED, "DROP_TIERED: remove object %s from metadata", name);
         WT_ERR_NOTFOUND_OK(__wt_metadata_remove(session, name), false);
+        filename = name;
+        WT_PREFIX_SKIP_REQUIRED(session, filename, "object:");
         if (remove_files && tier != NULL)
-            WT_TRET(__wt_meta_track_drop(session, tier->name));
+            WT_TRET(__wt_meta_track_drop(session, filename));
```

Comment:
Should we only be doing this when one of the calls to `wt_metadata_remove()`, above, was successful?

Reply:
Your comment prompted me to look closer. I have fixed it a different way since it was trying to drop all older objects and some may no longer exist on local storage which would make the meta track operation unhappy. I have added a more general call to `wt_fs_exist` to check for the older object locally and made it conditional on that.